junit-jupiter-api/junit-jupiter-api.gradle

```diff
@@ -2,7 +2,8 @@ description = 'JUnit Jupiter API'
 
 dependencies {
 	api("org.opentest4j:opentest4j:${ota4jVersion}")
-	api(project(":junit-platform-commons"))
+	api(project(':junit-platform-commons'))
+	compileOnly(project(':junit-platform-engine'))
```

Comment:
If we all agree to go this route, we should make sure `junit-platform-engine` is listed as `optional` in the generated POM.